src/Fractions/Fraction.cs

```diff
@@ -146,4 +153,180 @@ namespace Fractions;
+    }
+
+    static bool INumberBase<Fraction>.IsComplexNumber(Fraction value) {
+        return false;
+    }
+
+    static bool INumberBase<Fraction>.IsImaginaryNumber(Fraction value) {
+        return false;
+    }
+
+    static bool INumberBase<Fraction>.IsSubnormal(Fraction value) {
+        return false;
+    }
+
+    /// <summary>Determines if a value is infinite.</summary>
+    /// <param name="value">The value to be checked.</param>
+    /// <returns>
+    ///     <see langword="true" /> if <paramref name="value" /> is infinite; otherwise, <see langword="false" />.
+    /// </returns>
+    static bool INumberBase<Fraction>.IsInfinity(Fraction value) {
+        return value.IsInfinity;
+    }
+
+    /// <summary>Determines if a value is positive infinity.</summary>
+    /// <param name="value">The value to be checked.</param>
+    /// <returns>
+    ///     <see langword="true" /> if <paramref name="value" /> is positive infinity; otherwise, <see langword="false" />.
+    /// </returns>
+    static bool INumberBase<Fraction>.IsPositiveInfinity(Fraction value) {
+        return value.IsPositiveInfinity;
+    }
+
+    /// <summary>Determines if a value is negative infinity.</summary>
+    /// <param name="value">The value to be checked.</param>
+    /// <returns>
+    ///     <see langword="true" /> if <paramref name="value" /> is negative infinity; otherwise, <see langword="false" />.
+    /// </returns>
+    static bool INumberBase<Fraction>.IsNegativeInfinity(Fraction value) {
+        return value.IsNegativeInfinity;
+    }
+
+    /// <summary>Determines if a value is NaN.</summary>
+    /// <param name="value">The value to be checked.</param>
+    /// <returns>
+    ///     <see langword="true" /> if <paramref name="value" /> is NaN; otherwise, <see langword="false" />.
+    /// </returns>
+    static bool INumberBase<Fraction>.IsNaN(Fraction value) {
+        return value.IsNaN;
+    }
+#endif
+
+    /// <summary>
+    ///     Determines whether the given Fraction is in canonical form.
+    /// </summary>
+    /// <param name="value">The Fraction to check.</param>
+    /// <returns>True if the Fraction is in canonical form, otherwise false.</returns>
+    /// <remarks>
+    ///     A Fraction is considered to be in canonical form if its denominator is one,
+    ///     or if its numerator and denominator are coprime numbers (their greatest common divisor is one).
+    /// </remarks>
+    public static bool IsCanonical(Fraction value) {
+        var numerator = value.Numerator;
+        var denominator = value.Denominator;
+        if (denominator.IsOne) {
+            return true;
+        }
+
+        if (numerator.IsZero) {
+            return denominator.IsZero; // if we want to consider NaN as "canonical"
+        }
+
+        if (denominator.IsZero) {
+            return numerator.IsOne || numerator == BigInteger.MinusOne;
+        }
+
+        return BigInteger.GreatestCommonDivisor(numerator, denominator).IsOne;

```

Comment:
Can we optimize this here, or am I missing something?

```suggestion
        return !value._normalizationNotApplied ||
               BigInteger.GreatestCommonDivisor(numerator, denominator).IsOne;
```

Reply:
fixed in https://github.com/Fractions-DotNet/Fractions/pull/96/commits/1643913092e03ca145a91e4d5de95948f16949da